org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaDebugCache.scala

```diff
@@ -275,16 +318,7 @@ protected[debug] class ScalaDebugCacheActor(debugCache: ScalaDebugCache, debugTa
     }
   }
 
-  private def getLoadedNestedTypes(outerTypeName: String): Set[ReferenceType] = {
-    nestedTypesCache.get(outerTypeName) match {
-      case Some(cache) =>
-        cache.types
-      case None =>
-        initializedRequestsAndCache(outerTypeName).types
-    }
-  }
-
-  private def initializedRequestsAndCache(outerTypeName: String): NestedTypesCache = {
+  private def findNestedTypesWithJdi(outerTypeName: String) = {
```

Comment:
I'd rather put `jdiSynchronized` inside this method, rather than force all callers to call `jdiSynchronized` before calling this method.